From 64abeb43501fa24c897c4c2e0185888411d68864 Mon Sep 17 00:00:00 2001
From: Jouni Malinen <jouni@qca.qualcomm.com>
Date: Tue, 27 Mar 2012 18:04:06 +0300
Subject: [PATCH 60/73] P2P: Do not use group idle timeout during provisioning

Use the normal group formation timeout during the provisioning phase to
avoid terminating this process too early due to group idle timeout.

Signed-hostap: Jouni Malinen <jouni@qca.qualcomm.com>
(cherry picked from commit 8c472816fd2a234776356de8dc26f304376c48c6)
---
 wpa_supplicant/p2p_supplicant.c |   11 +++++++++++
 1 file changed, 11 insertions(+)

--- a/wpa_supplicant/p2p_supplicant.c
+++ b/wpa_supplicant/p2p_supplicant.c
@@ -3941,6 +3941,17 @@ static void wpas_p2p_set_group_idle_time
 	if (timeout == 0)
 		return;
 
+	if (wpa_s->p2p_in_provisioning) {
+		/*
+		 * Use the normal group formation timeout during the
+		 * provisioning phase to avoid terminating this process too
+		 * early due to group idle timeout.
+		 */
+		wpa_printf(MSG_DEBUG, "P2P: Do not use P2P group idle timeout "
+			   "during provisioning");
+		return;
+	}
+
 	wpa_printf(MSG_DEBUG, "P2P: Set P2P group idle timeout to %u seconds",
 		   timeout);
 	eloop_register_timeout(timeout, 0, wpas_p2p_group_idle_timeout,
